### DISCUSSION TITLE
calc() config file system

### DISCUSSION THREAD

**Discussion by thomasfarre** [tailwindlabs/tailwindcss#10075] [Help] [answered]
Hi everybody, I'm trying to find out my self but i'm having some trouble so if i can have some help it would be very cool ! 

I work at a company that use a design grid system to build everything (fluid size for columns and fixed for gutters) and a lot of times elements on a page are not exactly on a column.
So for now i use calc() fonction a lot to place things exactly where we need to depending on the size of our grid, number of colomns, gutters value etc...

![image](https://user-images.githubusercontent.com/71550958/207604965-d6899adb-9f59-43a8-a17f-3eb305467cf0.png)

In my case, with Tailwind JIT, my code looks like this 
-> example would be : 
w-[calc((100vw-100px)/12*2-30px)] 

- 100vw is screen size
- 100px is offset before grid start
- 12 is number of colomns in total
- 2 is number of colomn the element is taking
- -30px is because i need to move a bit the element and remove a gutter

It's working fine, but my problem is : 
-> It's a lot of code to write, painfull to read and maintain, hard to explain to others especially because with breakpoint we have often like 3 times this with different values in the same element.

I thought i could code something using Tailwind config file to simplify this, with my example above we could have :  "w-col-2/12" instead of calc(.....) but how can i keep the possibility to have the "-30px" has a variable? It could be -40px or 40px etc...

Do somebody have thoughts on this ? Can give me a hand ? It would be very appreciated !! 

Thanks, have a great day

**Comment by wongjn** [comment, marked as answer]
For the offset before the grid starts, you could use horizontal padding:

```html
<div class="px-[100px]">
  <!-- … -->
</div>
```

For the grid itself, you could use CSS grids (the gutter width was not specified so I have used `1rem` as an example):

```html
<div class="px-[100px] grid grid-cols-12 gap-x-4">
  <!-- … -->
</div>
```

For items to span columns, you could apply `grid-column`:

```html
<div class="px-[100px] grid grid-cols-12 gap-x-4">
  <div class="col-span-2">
    <!-- … -->
  </div>
</div>
```

**Reply by thomasfarre** [reply]
Hello, thanks for you'r answer
I'm familiar with grid but i found it not sufficient in my case because a lot of times i have every elements of my page to put on it and i can't put everything on columns. Sometimes, elements are between gutters for example

**Reply by wongjn** [reply]
You could simply not use a grid system in those instances.

**Reply by thomasfarre** [reply]
It's not me who design things, i need to follow it

**Reply by wongjn** [reply]
Indeed, You can follow it by not using CSS grid in those instances.

**Reply by thomasfarre** [reply]
Like i said in my first post, i know how to make things work right now. Using calc or grid it will work, it's not really my question. 
To resume, i was wondering if we could create a "system" to avoid writing long sentence like this one : w-[calc((100vw-100px)/12*2-30px)]

**Reply by wongjn** [reply]
Yes indeed you could. You could generate additional `spacing` or `width` values (programmatically or otherwise) or a plugin in the `tailwind.config.js`.

**Reply by thomasfarre** [reply]
Ok i think i understand, inside my tailwind.config file i can write like

width:  {
 'col-1/12': 'calc((100vw - 100px) / 12 * 2)'
}

Then inside my html template i can write :

w-[calc(theme('width[col-1/12]')-30px)]

Something like that ?

**Reply by wongjn** [reply]
Yes.
